Sources/Swift/Helper/SentryDataWrapper.swift

```diff
@@ -0,0 +1,605 @@
+        self.data.resetBytes(in: range)
+    }
+
+    /// Replace a region of bytes in the data with new data.
+    ///
+    /// This will resize the data if required, to fit the entire contents of `SentryDataWrapper`.
+    ///
+    /// - precondition: The bounds of `subrange` must be valid indices of the collection.
+    /// - parameter subrange: The range in the data to replace. If `subrange.lowerBound == data.count && subrange.count == 0` then this operation is an append.
+    /// - parameter data: The replacement data.
+    public mutating func replaceSubrange(_ subrange: Range<Data.Index>, with data: Data) {
+        self.data.replaceSubrange(subrange, with: data)
+    }
+
+    /// Replace a region of bytes in the data with new bytes from a buffer.
+    ///
+    /// This will resize the data if required, to fit the entire contents of `buffer`.
+    ///
+    /// - precondition: The bounds of `subrange` must be valid indices of the collection.
+    /// - parameter subrange: The range in the data to replace.
+    /// - parameter buffer: The replacement bytes.
+    public mutating func replaceSubrange<SourceType>(_ subrange: Range<Data.Index>, with buffer: UnsafeBufferPointer<SourceType>) {
+        self.data.replaceSubrange(subrange, with: buffer)
+    }
+
+    /// Replace a region of bytes in the data with new bytes from a collection.
+    ///
+    /// This will resize the data if required, to fit the entire contents of `newElements`.
+    ///
+    /// - precondition: The bounds of `subrange` must be valid indices of the collection.
+    /// - parameter subrange: The range in the data to replace.
+    /// - parameter newElements: The replacement bytes.
+    public mutating func replaceSubrange<ByteCollection>(_ subrange: Range<Data.Index>, with newElements: ByteCollection) where ByteCollection: Collection, ByteCollection.Element == UInt8 {
+        self.data.replaceSubrange(subrange, with: newElements)
+    }
+
+    public mutating func replaceSubrange(_ subrange: Range<Data.Index>, with bytes: UnsafeRawPointer, count cnt: Int) {
+        self.data.replaceSubrange(subrange, with: bytes, count: cnt)
+    }
+
+    /// Return a new copy of the data in a specified range.
+    ///
+    /// - parameter range: The range to copy.
+    public func subdata(in range: Range<Data.Index>) -> Data {
+        return self.data.subdata(in: range)
+    }
+
+    /// Write the contents of the `SentryDataWrapper` to a location.
+    ///
+    /// - parameter url: The location to write the data into.
+    /// - parameter options: Options for writing the data. Default value is `[]`.
+    /// - throws: An error in the Cocoa domain, if there is an error writing to the `URL`.
+    public func write(to url: URL, options: Data.WritingOptions = []) throws {
+        // TODO: begin file.write via static tracker
+        try self.data.write(to: url, options: options)
+        // TODO: end file.write via static tracker
+    }
```

Comment:
@philipphofmann @brustolin @armcknight 

This is my draft for a wrapper for `Swift.Data`, as it does not bridge to NSData anymore.

I created a Swift struct that can mostly be used as a direct drop-in replacement for `Swift.Data` (it's a `@frozen struct` so I don't think we can "subclass"/swizzle/monkey-patch). 
I also considered creating this wrapper as an Objective-C file, if I am not mistaken this would require to use `NSData`, because AFAIK we can not use the `Swift.Data` struct from Objective-C classes.

Is this the right approach? Right now I am not sure how to access the hub from here and create the `file.read/write` spans. If we want to mimic the swizzling in `SentryNSDataSwizzling.m`, we would need access to the `SentryNSDataSwizzling.shared.dataTracker` or similar

Reply:
Closing this comment, further discussion in #4634

Reply:
Imitating Sentry.Data does sound like a good solution, but it requires a lot of work.

Since we only track the atomic reads and writes of NSData, Im wondering if we could of a wrapper and extensions for those functions alone and then return a Sentry.Data

Let's say SentryData.dataForUrl(:) (and similar) returns a Data object. 
And a some Data extensions like .trackingWrite(path:).

But this is not a drop-in replacement and could discourage usage, so if you think maintaining a complete copy of it won't be a problem I believe we can go with it.

Reply:
Yes, I'm also wondering if we can do something better than a copy. Maybe we should instead come up with a better API to add custom spans to transactions bound to the scope in general. Or maybe extension methods. I'm not sure yet.